File: workspace/skills/asaas/scripts/asaas_client.js
const fs = require('fs');
const path = require('path');

const CONFIG_PATH = '/Users/mauricio/openclaw/config/asaas.json';

function loadConfig() {
  return JSON.parse(fs.readFileSync(CONFIG_PATH, 'utf8'));
}

async function request(endpoint, params = {}) {
  const config = loadConfig();
  const url = new URL(`${config.base_url}${endpoint}`);
  Object.entries(params).forEach(([k, v]) => {
    if (v) url.searchParams.set(k, v);
  });

  const response = await fetch(url.toString(), {
    headers: {
      'access_token': config.api_key,
      'Content-Type': 'application/json'
    }
  });

  if (!response.ok) {
    throw new Error(`Asaas API Error: ${response.status} ${response.statusText}`);
  }

  return response.json();
}

async function listCustomers(identifier) {
  const params = {};
  if (identifier.includes('@')) {
    params.email = identifier;
  } else {
    params.cpfCnpj = identifier.replace(/\D/g, '');
  }
  return request('/customers', params);
}

async function listPayments(customerId) {
  return request('/payments', { customer: customerId, limit: 20 });
}

async function main() {
  const args = process.argv.slice(2);
  const cmd = args[0];
  const arg1 = args[1];

  if (!cmd) {
    console.log("Usage: node asaas_client.js <search_customer|get_payments|check_student> <arg>");
    return;
  }

  try {
    if (cmd === 'search_customer') {
      console.log(JSON.stringify(await listCustomers(arg1), null, 2));
    } else if (cmd === 'get_payments') {
      console.log(JSON.stringify(await listPayments(arg1), null, 2));
    } else if (cmd === 'check_student') {
      const customers = await listCustomers(arg1);
      if (!customers.data || customers.data.length === 0) {
        console.log(JSON.stringify({ error: "Customer not found" }, null, 2));
        return;
      }
      const customer = customers.data[0];
      const payments = await listPayments(customer.id);
      
      console.log(JSON.stringify({
        customer: {
            name: customer.name,
            id: customer.id,
            email: customer.email,
            cpfCnpj: customer.cpfCnpj
        },
        payments: payments.data || []
      }, null, 2));
    }
  } catch (e) {
    console.error(e.message);
  }
}

main();

File: skills/asaas/scripts/asaas_client.py
import os
import json
import sys
import requests

CONFIG_PATH = '/Users/mauricio/openclaw/config/asaas.json'

def load_config():
    with open(CONFIG_PATH, 'r') as f:
        return json.load(f)

def get_headers(api_key):
    return {
        'access_token': api_key,
        'Content-Type': 'application/json'
    }

def list_customers(email=None, cpf=None):
    config = load_config()
    url = f"{config['base_url']}/customers"
    params = {}
    if email: params['email'] = email
    if cpf: params['cpfCnpj'] = cpf
    
    response = requests.get(url, headers=get_headers(config['api_key']), params=params)
    return response.json()

def list_payments(customer_id=None):
    config = load_config()
    url = f"{config['base_url']}/payments"
    params = {'limit': 20}
    if customer_id: params['customer'] = customer_id
    
    response = requests.get(url, headers=get_headers(config['api_key']), params=params)
    return response.json()

def main():
    if len(sys.argv) < 2:
        print("Usage: python asaas_client.py <command> [args]")
        sys.exit(1)
        
    cmd = sys.argv[1]
    
    if cmd == 'search_customer':
        email = sys.argv[2] if '@' in sys.argv[2] else None
        cpf = sys.argv[2] if '@' not in sys.argv[2] else None
        print(json.dumps(list_customers(email, cpf), indent=2))
        
    elif cmd == 'get_payments':
        customer_id = sys.argv[2]
        print(json.dumps(list_payments(customer_id), indent=2))
        
    elif cmd == 'check_student':
        # Search by email/cpf and get payments
        identifier = sys.argv[2]
        email = identifier if '@' in identifier else None
        cpf = identifier if '@' not in identifier else None
        
        customers = list_customers(email, cpf)
        if not customers.get('data'):
            print(json.dumps({"error": "Customer not found"}, indent=2))
            return
            
        customer = customers['data'][0]
        payments = list_payments(customer['id'])
        
        result = {
            "customer": {
                "name": customer['name'],
                "id": customer['id'],
                "email": customer['email'],
                "cpfCnpj": customer['cpfCnpj']
            },
            "payments": payments.get('data', [])
        }
        print(json.dumps(result, indent=2))

if __name__ == "__main__":
    main()
